_posts/2020-05-29-fault-tolerance-4-2-and-4-3.adoc

```diff
@@ -0,0 +1,72 @@
+:page-layout: post
+:page-title: Fault Tolerance 4.2 & 4.3
+:page-synopsis: SmallRye Fault Tolerance 4.2 and 4.3 released!
+:page-tags: [announcement, microprofile]
+:page-date: 2020-05-29 12:00:00.000 +0100
+:page-author: lthon
+
+= Fault Tolerance 4.2 & 4.3
+
+Today, we announce two new releases of SmallRye Fault Tolerance: 4.2.1 and 4.3.0.
+
+== 4.2.1
+
+This release includes bugfixes and component updates and is safe to use for everyone who uses Fault Tolerance 4.1.0, 4.1.1 and 4.2.0.
+The 4.2.0 release happened a few weeks ago and didn't contain anything but bugfixes either.
+With 4.2.1, we now use MicroProfile Fault Tolerance 2.1.1, which includes some TCK fixes.
+
+There is one small feature in the 4.2.1 release, which is interesting for integrators.
+If you integrate SmallRye Fault Tolerance and find you need to provide your own `ExecutorFactory` because you need SmallRye Fault Tolerance to use different thread pools, you can now inherit from `DefaultExecutorFactory` and override the `threadFactory` method.
+This allows you to customize the `ThreadFactory` all the thread pools use, while retaining all the other default settings.
+
+== 4.3.0
+
+This release builds on 4.2.1 and adds one new feature: you can now observe when circuit breakers change their state.
+Let's first take a look at the code.
+
+Let's say we have a method `doWork` which is guarded by a circuit breaker, presumably because it reaches out to some external service:
+
+[source,java]
+----
+@ApplicationScoped
+public class MyService {
+    @CircuitBreaker(...)
+    public void doWork() {
+        ...
+    }
+}
+----
+
+Now, there are metrics for how often the circuit breaker trips etc., but you might want to observe the circuit breaker state changes directly.
```

Comment:
`want to observe the circuit breaker` replace by `want to observe when the circuit breaker´?

Reply:
Both are possible, but I guess adding "when" is probably better.